notes.md

```diff
@@ -0,0 +1,124 @@
+* The date the share was purchased is a property of the block the purchase was processed in so there is no need to store it in the token. However the user database might wish to record that information as well for convenience.
+* The price the user paid is a function of the ETH price per Token (which can change during the crowdsale) and ETH price at that point (which can also be set in the `CrowdsaleContract`) and the number of ETH spent, which the `CrowdsaleContract` also knows and which is returned in the transaction receipt.
+* With regard to the certificate number, the token should record the certificate numbers, either algorithmically if possible, or else it will need to be generated offchain and imported in bulk when the certificates themselves are issued.
+* A certificate might represent more than one share, in which case it will need to record a range of inputs in the form of an encoded bytes array.
+* as shares are traded a script needs to pull Transfer events from the global blockchain and keep the off-chain records up to date, either in real-time, or in an ad-hoc way.  This should be the job of the Crowdsale Admin website.
+
+### Questions and Scenarios.
+
+1. When someone buys a Token does that mean they have bought a specific Stock with a specific certificate number, or does it mean they've bought the right to be issued a stock certificate?
+
+    scenario: Token Buys Share immediately.
+    scenario: Token Buys rights to a share, when the crowdsale is finished.
+
+2. If I buy 100 Tokens would that result in the issuance of 100 certificates, or one certificate with 100 shares on it.
+
+    scenario: One token = one certificate = one share.
+    scenario: One token = one certificate has many shares.
+
+3. Certificates will take their numbers from what's in the blockchain or else will use numbers generated externally.
+
+    scenario: Certificate numbers come from the blockchain.
+    scenatio: Certificate numbers are generated externally.
+
+a) `Token Buys Share immediately` && `One token = one certificate = one share` && `Certificate numbers come from the blockchain`
+
+* Lift most of `ERC721` but override the `transfer`, `transferFrom` and `mint` functions so they retain the `value` params from `ERC20`, and under the hood behave like ERC721 tokens instead, to a `transfer(0x..., 10)` would transfer 10 Tokens.  This should be enough to retain compatability with ERC20.
+* A person's address sends ETH to the `CrowdsaleContract` and a `Token` is minted for them.
+* Each `Token` needs to be issued one at a time by the Crowdsale visa the `mint` function.
+* When a `Token` is minted for an address, the `tokenId`, ie a sequential issuance number starting from some `issuanceOffset`, is associated with that address. Then we can say, some time in the future, that address `0xnnnnn` owns Tokens `1`, `5`, and `33` (lets say)
```

Comment:
It might be easier to think of it as who owns share 1, who owns share 5, who owns share 33 instead of having a smart-contract track this.

Reply:
Maybe there is an upgradeable stock ledger contract which tracks all the shares and their state, say 150,000 and has a hard coded range for this sale from 0 - 15000?

Reply:
I've updated the doc to cover this off.

Reply:
@davesag can tokens be minted into an address owned by the company as "unissued" and then be issued as tokens are paid?